aea/skills/tasks.py

```diff
@@ -18,72 +18,102 @@
+        :param args: the positional arguments to be passed to the function.
+        :param kwds: the keyword arguments to be passed to the function.
+        :return the task id to get the the result.
+        :raises ValueError: if the task manager is not running.
         """
-        self.task_queue.put(task)
-
-    def dispatch(self) -> None:
-        """Dispatch a task."""
-        while not self.stopped:
-            for future in self.futures:
-                if future.done():
-                    future.result()
-
-            next_task = self.task_queue.get(block=True)  # type: Optional[Task]
-            if next_task is None:
-                return
-
-            future = self._executor.submit(next_task.execute)
-            self.futures.append(future)
+        with self._lock:
+            if self._stopped:
+                raise ValueError("Task manager not running.")
+            self._pool = cast(Pool, self._pool)
+            task_id = self._task_enqueued_counter
+            self._task_enqueued_counter += 1
+            async_result = self._pool.apply_async(
+                func, args=args, kwds=kwds if kwds is not None else {}
+            )
+            self._results_by_task_id[task_id] = async_result
+            return async_result
+
+    def get_task_result(self, task_id: int) -> AsyncResult:
+        """Get the result from a task."""
+        task_result = self._results_by_task_id.get(
+            task_id, default=None
+        )  # type: Optional[AsyncResult]
+        if task_result is None:
+            raise ValueError("Task id {} not present.".format(task_id))
+
+        return task_result
 
     def start(self) -> None:
         """Start the task manager."""
-        with self.lock:
-            logger.debug("Start the task manager.")
-            self.stopped = False
-            self.thread.start()
+        with self._lock:
+            if self._stopped is False:
+                logger.debug("Task manager already running.")
+            else:
+                logger.debug("Start the task manager.")
+                self._stopped = False
+                self._pool = multiprocessing.Pool(
+                    self._nb_workers, initializer=init_worker
+                )
 
     def stop(self) -> None:
         """Stop the task manager."""
-        with self.lock:
-            if self.stopped is True:
-                return
-            logger.debug("Stop the task manager.")
-            self.stopped = True
-            self.task_queue.put(None)
-            self.thread.join()
-            self.thread = Thread(target=self.dispatch)
+        with self._lock:
+            if self._stopped is True:
+                logger.debug("Task manager already stopped.")
+            else:
+                logger.debug("Stop the task manager.")
+                self._stopped = True
+                self._pool = cast(Pool, self._pool)
+                self._pool.terminate()
```

Comment:
this is not a graceful termination because we overwrote the SIGINT handler. See above.